src/main/java/com/google/devtools/build/lib/rules/cpp/CcCommon.java

```diff
@@ -1036,7 +1038,9 @@ public static FeatureConfiguration configureFeaturesOrThrowEvalException(
       }
     }
 
-    allFeatures.addAll(getCoverageFeatures(cppConfiguration));
+    if (ruleContext == null || CcCompilationHelper.isCodeCoverageEnabled(ruleContext)) {
```

Comment:
Does it make sense to permit ruleContext to be null? The semantics in this case are not intuitive to me.
Is there even a context where this gets called from that doesn't have access to a rule context?